src/Utilities/TicToc/TicToc.jl

```diff
@@ -144,16 +144,21 @@ end
 """
     tictoc()
 
-Call at program start to set up the globals used by the macros and
-to register the at-exit callback.
+Call at program start (only once!) to set up the globals used by the
+macros and to register the at-exit callback.
 """
 function tictoc()
     global timing_info_names
     for nm in timing_info_names
         exti = Symbol(nm)
-        global ex = Expr(:global, Expr(:(=), exti, Expr(:call, Expr(:(.), :TicToc, QuoteNode(:TimingInfo)))))
-        eval(ex)
-        eval(Expr(:call, :push!, :timing_infos, exti))
+        if isdefined(@__MODULE__, exti)
```

Comment:
This means that repeated calls to `tictoc` will fail. I rather check that only one entity is declaring a global. E.g.only one use of `@tic name`